frontend-build-pipeline.yml

```diff
@@ -39,53 +39,124 @@ parameters:
+          extraProperties: |
+            sonar.coverage.exclusions=${{ parameters.sonarCloudCoverageExclude }}
+            sonar.javascript.lcov.reportPaths=${{ parameters.sonarCloudCoverageDir }}
+
+      - task: SonarCloudAnalyze@1
+
+      - task: SonarCloudPublish@1
+        inputs:
+          pollingTimeoutSec: '300'
+
+      - task: PublishPipelineArtifact@1
+        inputs:
+          targetPath: '$(Build.Repository.LocalPath)/dist'
+          artifact: 'front-end-artifact'
+          publishLocation: 'pipeline'
+        displayName: Publish Pipeline Artifact
+
+      # Docker Image is created and pushed to ACR with build specific tag.
+
+      - template: "./azure-templates/docker-template.yml"
+        parameters:
+          repository: frontend
+          tag: "$(Build.BuildNumber)"
+          Dockerfile: ./Dockerfile
+          containerRegistry: "acr-team3"
+
+      # Docker Image is created and pushed to ACR with latest tag.
+
+      - template: "./azure-templates/docker-template.yml"
+        parameters:
+          repository: frontend
+          tag: "latest"
+          Dockerfile: ./Dockerfile
+          containerRegistry: "acr-team3"
+
+
+# Deployment Stage. Using images pushed to the Azure Container Registry, and a helm Chart.
+
+- stage: Deploy
+  displayName: Deploy to AKS Cluster
+  dependsOn: TestBuild
+  condition: and(succeeded(), in(variables['Build.Reason'], 'IndividualCI', 'BatchedCI'))
+  variables:
+    releaseNumber: $[counter('swag-release.1.', 100)]
+  jobs:
+    - job: HelmDeploy
+      steps:
+
+        # In the next steps, we're dynamically determining Deployment Environment based on branch of CI. 
+        # If not main, staging, or development, ideally this would be test.
+        # For testing purposes the default is dev. To be updated when a production and testing environment are available..
+
+        # Creating Image Pull secrets for Helm/K8S to access the images and use on the cluster.
+
+        - task: KubernetesManifest@0
+          displayName: Create Pull Secret Dev
+          inputs:
+            action: "createSecret"
+            ${{ if eq(variables['Build.SourceBranchName'], 'development') }}:
+              kubernetesServiceConnection: "k8s-team3-config-dev"
+            ${{ if eq(variables['Build.SourceBranchName'], 'staging') }}:
+              kubernetesServiceConnection: "k8s-team3-config-staging"
+            ${{ if eq(variables['Build.SourceBranchName'], 'feature_cd') }}:
+              kubernetesServiceConnection: "k8s-team3-config-dev"
+            namespace: "team3"
+            secretType: "dockerRegistry"
+            secretName: "team3pullsecret"
+            dockerRegistryEndpoint: "acr-team3"
+
+        # Helm deploys to a cluster, dependant on branch. development = dev, staging = staging (main = production)
+
+        - template: "./azure-templates/helm-template.yml"
+          parameters:
+            ${{ if eq(variables['Build.SourceBranchName'], 'development') }}:
```

Comment:
note that Build.SourceBranchName breaks on branches that contain slashes in their names (like `feature/abc`), so it should not be used in general, but it will work for now

Reply:
Changed to using Build.SourceBranch to avoid this issue.